### PR TITLE
Change the GetLogger function from Mutex to RWMutex

### PR DESCRIPTION
**Pull request by tangyouyou**
Change the GetLogger function from Mutex to RWMutex

### CODE REVIEW

**Review by chenhengqi — COMMENTED** [review]
What's the point of this change?

**Comment by tangyouyou** [comment]
> What's the point of this change?

RWMutex is faster than Mutex, See pictures for more details.
RMMutex

**Comment by chenhengqi** [comment]
> > What's the point of this change?
> 
> RWMutex is faster than Mutex, See pictures for more details.
> RMMutex

Performance isn't an issue in this case.

File: tars/util/rogger/logger.go [hunk 1]
```diff
@@ -92,17 +92,23 @@ func (lv *LogLevel) coloredString() string {
...
-	defer loggerMutex.Unlock()
-	if lg, ok := loggerMap[name]; ok {
-		return lg
-	}
-	lg := &Logger{
-		name:   name,
-		writer: &ConsoleWriter{},
+	var (
+		lg *Logger
+		ok = false
```
Comment (chenhengqi) [inline tars/util/rogger/logger.go:97]: use `var ok bool` instead

File: tars/util/rogger/logger.go [hunk 2]
```diff
@@ -92,17 +92,23 @@ func (lv *LogLevel) coloredString() string {
...
+	)
+	loggerMutex.RLock()
+	if lg, ok = loggerMap[name]; ok == false {
```
Comment (chenhengqi) [inline tars/util/rogger/logger.go:100]: use `!ok` instead

File: tars/util/rogger/logger.go [hunk 3]
```diff
@@ -92,17 +92,23 @@ func (lv *LogLevel) coloredString() string {
...
+		loggerMutex.RUnlock()
+		loggerMutex.Lock()
+		lg = &Logger{
+			name:   name,
```
Comment (chenhengqi) [inline tars/util/rogger/logger.go:104]: missing the `writer` field

**Review by chenhengqi — CHANGES_REQUESTED** [review]

**Comment by defool** [comment]
应该在初始化时GetLogger，而不是每次打日志都调用，这样的话性能不是问题。
先关闭这个pr，如果有需要可以重新打开。

**Pull request closed without merging**